src/index.ts

```diff
@@ -0,0 +1,80 @@
+import { TASK_COMPILE_GET_SOURCE_PATHS } from "@nomiclabs/buidler/builtin-tasks/task-names";
+import { task } from "@nomiclabs/buidler/config";
+import { BuidlerConfig } from "@nomiclabs/buidler/types";
+import { ensureDir, readFile, writeFile } from "fs-extra";
+import { defaults } from "lodash";
+import path from "path";
+import { processCode } from "solpp";
+
+declare module "@nomiclabs/buidler/types" {
+  interface BuidlerConfig {
+    solpp?: {
+      defs?: string;
+      cwd?: string;
+      name?: string;
+      collapseEmptyLines?: boolean;
+      noPreprocessor?: boolean;
+      noFlatten?: boolean;
+      tolerant?: boolean;
+    };
+  }
+}
+
+export const PROCESSED_CACHE_DIRNAME = "solpp-generated-contracts";
+
+function getDefaultConfig(config: BuidlerConfig) {
+  let sourcesPath = process.cwd();
+  if (config.paths !== undefined && config.paths.sources !== undefined) {
+    sourcesPath = config.paths.sources;
+  }
+  return {
+    defs: {},
+    cwd: sourcesPath,
+    name: "buidler-solpp",
+    collapseEmptyLines: false,
+    noPreprocessor: false,
+    noFlatten: false,
+    tolerant: false
+  };
+}
+
+function getConfig(config: BuidlerConfig) {
+  return defaults(config.solpp, getDefaultConfig(config));
+}
+
+function readFiles(filePaths: string[]) {
+  return Promise.all(
+    filePaths.map(filePath =>
+      readFile(filePath, "utf-8").then(content => [filePath, content])
+    )
+  );
+}
+
+task("buidler-solpp:run-solpp", async ({ files, opts }, { config, run }) => {
```

Comment:
`internalTask`? Aren't `files` and `opts` implicit `any`s? I don't get how this works

Reply:
Anyway, the params should be defined.